core/core.go

```diff
@@ -382,33 +382,76 @@ func makeAddrsFactory(cfg config.Addresses) (p2pbhost.AddrsFactory, error) {
 	}, nil
 }
 
-func makeSmuxTransport(mplexExp bool) smux.Transport {
+func makeSmuxTransport(multiplexers []string, mplexExp bool) smux.Transport {
 	mstpt := mssmux.NewBlankTransport()
 
-	ymxtpt := &yamux.Transport{
-		AcceptBacklog:          512,
-		ConnectionWriteTimeout: time.Second * 10,
-		KeepAliveInterval:      time.Second * 30,
-		EnableKeepAlive:        true,
-		MaxStreamWindowSize:    uint32(1024 * 512),
-		LogOutput:              ioutil.Discard,
+	multiplexerProtocolNames := map[string]string{
+		"yamex": "/yamux/1.0.0",
+		"mplex": "/mplex/6.7.0",
 	}
 
-	if os.Getenv("YAMUX_DEBUG") != "" {
-		ymxtpt.LogOutput = os.Stderr
+	var isYamex, isMplex bool
+
+	for _, multiplexer := range multiplexers {
+		if multiplexer == "yamex" {
+			isYamex = true
+		} else if multiplexer == "mplex" {
+			isMplex = true
+			if mplexExp {
+				log.Warning("--enable-mplex-experiment is set to false, " +
+					"but mplex is given as a multiplexer in the config.")
+			}
+		} else {
+			log.Warningf("Unknown multiplexer specified in config: %s", multiplexer)
```

Comment:
nit: I'd turn these warnings into errors. That way, they'll get printed to stdout on daemon start (we'd like to print warnings as well but they're too noisy at the moment).